### ISSUE TITLE
Multiple threshold for Multilabel metrics

### ISSUE THREAD

**Issue by ding3820** [Lightning-AI/torchmetrics#2612]
## 🚀 Feature

Allow multiple thresholds for every multilabel metrics

### Motivation

I notice that for every multilabel metrics, threshold only allows a float. In some cases, we would like to use different threshold for different class. 

### Pitch
For example: 
`mla = MultilabelAccuracy(threshold=[0.3, 0.7, 0.55], num_labels=3, average=None)`

**Comment by SkafteNicki** [comment]
Hi @ding3820, thanks for raising this issue.
We have evaluated that it will require a large refactor of the codebase to support this feature and we are therefore not going to support it in the near future. If this get requested in the future we may reconsider.
For now, there is a fairly easy way to achieve this in torchmetrics using `MetricCollection`:
```python
from torchmetrics import MetricCollection
from torchmetrics.classification import MultilabelAccuracy
import torch

mla = MetricCollection(
    {f"accuracy_{i}": MultilabelAccuracy(num_labels=3, average=None, threshold=t) for i, t in enumerate([0.1, 0.5, 0.9])}
)
print(mla)
#MetricCollection(
#  (accuracy_0): MultilabelAccuracy()
#  (accuracy_1): MultilabelAccuracy()
#  (accuracy_2): MultilabelAccuracy()
#)
x = torch.rand(10, 3)
y = torch.randint(0, 2, (10, 3))
mla.update(x, y)
print(mla.compute())
# {'accuracy_0': tensor([0.5000, 0.4000, 0.4000]), 
# 'accuracy_1': tensor([0.7000, 0.4000, 0.5000]), 
# 'accuracy_2': tensor([0.5000, 0.5000, 0.4000])}
```
it is a bit slower than if this was implemented directly into the metric but it should still work. Closing issue for now.
